api/app/users/tests.py

```diff
@@ -472,6 +422,264 @@ def test_get_comments(self):
+        self.event1_application_form = self.create_application_form(self.event1.id, True, False)
+        self.event2_application_form = self.create_application_form(self.event2.id, True, False)
+        db.session.add_all([self.event1_application_form, self.event2_application_form])
+
+        db.session.flush()
+
+        self.event1_id = self.event1.id  #TODO: add comment about why we do this
+
+    def test_event_does_not_exist(self):
+        """
+        Test that filtering on a nonexistent event 403s. It 403s because the user does
+        not have the required auth which is checked before the event is searched for.
+        """
+        self.seed_static_data()
+
+        # Make the request
+        header = self.get_auth_header_for(self.event1_admin.email)
+        params = {'event_id': 1337}
+
+        response = self.app.get('/api/v1/userprofilelist', headers=header, data=params)
+
+        self.assertEqual(response.status_code, 403)
+
+    def test_is_not_active_or_deleted(self):
+        """
+        Test that inactive or deleted users are not returned.
+        """
+        self.seed_static_data()
+
+        # Setup the candidates
+        candidates = self.add_n_users(3, organisation_id=self.dummy_org_id)
+        candidates[0].active = False
+        candidates[1].is_deleted = True
+        candidates[2].active = False
+        candidates[2].is_deleted = True
+
+        db.session.add_all(candidates)
+        db.session.flush()
+
+        responses = [
+            Response(self.event1_application_form.id, candidates[0].id, True),
+            Response(self.event1_application_form.id, candidates[1].id, True),
+            Response(self.event1_application_form.id, candidates[2].id, True)
+        ]
+        db.session.add_all(responses)
+        db.session.commit()
+
+        # Make the request
+        header = self.get_auth_header_for(self.event1_admin.email)
+        params = {'event_id': self.event1_id}
+
+        response = self.app.get('/api/v1/userprofilelist', headers=header, data=params)
+
+        # Assert stuff
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json, [])
+
+    def test_no_response_and_not_invited(self):
+        """
+        Users that are not invited and have not responded should not be returned.
+        """
+        self.seed_static_data()
+
+        candidates = self.add_n_users(3, organisation_id=self.dummy_org_id)
+        db.session.add_all(candidates)
+        db.session.commit()
+
+        # Make the request
+        header = self.get_auth_header_for(self.event1_admin.email)
+        params = {'event_id': self.event1_id}
+
+        response = self.app.get('/api/v1/userprofilelist', headers=header, data=params)
+
+        # Assert that request succeeds and no users are returned.
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json, [])
+
+    def test_response_and_not_invited(self):
```

Comment:
This seems to be testing the same thing as `test_response_and_candidate` below, but that's doing it better, so rather remove this one.